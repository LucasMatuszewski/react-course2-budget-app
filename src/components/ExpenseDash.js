import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDash = () => (
    <div>
        <h2>Expenses Dashboard</h2>

        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList /> {/* We don't need to pass any data/props here. connect() will pass them directly from a store / state tree */}
        <br />
        <hr />
        <a href="/help">Server Side Rendering HTML Link</a><br />
        (html "a" tag will reload a browser, we can use it to link to external page)
    </div>
);

export default ExpenseDash;