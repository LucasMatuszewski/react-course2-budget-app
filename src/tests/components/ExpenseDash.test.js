import React from 'react';
import { shallow } from 'enzyme';
import ExpenseDash from '../../components/ExpenseDash';

test('should render Expenses Dashboard with Filters and List components', () => {
    const wrapper = shallow(<ExpenseDash />);
    expect(wrapper).toMatchSnapshot();
});
