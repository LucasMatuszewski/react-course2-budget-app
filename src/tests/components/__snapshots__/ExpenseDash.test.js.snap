// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expenses Dashboard with Filters and List components 1`] = `
<div>
  <h2>
    Expenses Dashboard
  </h2>
  <Connect(ExpenseListFilters) />
  <Connect(ExpenseList) />
   
  <br />
  <hr />
  <a
    href="/help"
  >
    Server Side Rendering HTML Link
  </a>
  <br />
  (html "a" tag will reload a browser, we can use it to link to external page)
</div>
`;
